src/metabase/api/embed.clj

```diff
@@ -235,7 +235,8 @@
   [& {:keys [export-format card-id embedding-params token-params query-params qp-runner constraints options]
       :or   {qp-runner qp/process-query-and-save-execution!}}]
   {:pre [(integer? card-id) (u/maybe? map? embedding-params) (map? token-params) (map? query-params)]}
-  (let [merged-slug->value (validate-and-merge-params embedding-params token-params (normalize-query-params query-params))
+  (let [merged-slug->value (-> (validate-and-merge-params embedding-params token-params (normalize-query-params query-params))
```

Comment:
we need test for this. and wdyt if we move the `update-vals` into `validate-and-merge-params` instead?

Reply:
Added tests in the embed_test namespace, thanks for recommending this because it caught a bug. 

I added `update-vals` to the `normalize-query-params` function, where it belonged in the first place. Because it was only needed to clean up query-params.